docs/core/diagnostics/eventsource-activity-ids.md

```diff
@@ -0,0 +1,287 @@
+Long ago a typical application may have been simple and single-threaded which makes logging straightforward. We
+could write each step to a log file in order and then read the log back exactly in the order it was written to
+understand what happened. If the app handled requests then being single threaded only one request was handled
+at a time. All log messages for request A would be printed in order, then all the messages for B, and so on.
+When apps become multi-threaded that strategy no longer works because multiple
+requests are being handled at the same time. However if each request is assigned to a single thread which processes
+it entirely we can solve the problem by recording a thread id for each log message. For example a multi-threaded app
+might log:
+
+```
+Thread Id      Message
+---------      -------
+12             BeginRequest A
+12             Doing some work
+190            BeginRequest B
+12             uh-oh error happened
+190            Doing some work
+```
+
+By reading the thread ids we know that thread 12 was processing request A and thread 190 was processing request B,
+therefore the 'uh-oh error happened' message related to request A. However application concurrency has continued to
+grow ever more sophisticated. It is now common to use async/await so that a single request could
+be handled partially on many different threads before the work is complete. Thread ids no longer are sufficient to
+correlate together all the messages produced for one request. Activity IDs solve this problem. They provide a finer
+grain identifier that can track individual requests, or portions of requests, regardless if the work
+is spread across different threads.
+
+> [!NOTE]
+> The Activity ID concept refered to here is not the same as the System.Diagnostics.Tracing.Activity, despite the
+> similar naming.
+
+## Tracking work using an Activity ID
+
+We can run the code below to see Activity tracking in action.
+
+```C#
+using System.Diagnostics.Tracing;
+
+public static class Program
+{
+    public static async Task Main(string[] args)
+    {
+        ConsoleWriterEventListener listener = new ConsoleWriterEventListener();
+
+        Task a = ProcessWorkItem("A");
+        Task b = ProcessWorkItem("B");
+        await Task.WhenAll(a, b);
+    }
+
+    private static async Task ProcessWorkItem(string requestName)
+    {
+        DemoEventSource.Log.WorkStart(requestName);
+        await HelperA();
+        await HelperB();
+        DemoEventSource.Log.WorkStop();
+    }
+
+    private static async Task HelperA()
+    {
+        DemoEventSource.Log.DebugMessage("HelperA");
+        await Task.Delay(100); // pretend to do some work
+    }
+
+    private static async Task HelperB()
+    {
+        DemoEventSource.Log.DebugMessage("HelperB");
+        await Task.Delay(100); // pretend to do some work
+    }
+}
+
+[EventSource(Name ="Demo")]
+class DemoEventSource : EventSource
+{
+    public static DemoEventSource Log = new DemoEventSource();
```

Comment:
Either
```suggestion
    public static readonly DemoEventSource Log = new DemoEventSource();
```
or
```suggestion
    public static DemoEventSource Log { get; } = new DemoEventSource();
```
to have it as static property.

The same in other code examples showing an `EventSource`.

Reply:
Updated all the examples to use a getter